### PR TITLE
Add support for gradesReleased LTI flag

### PR DESCRIPTION
**Pull request by Greyvend**
We need this flag support to implement [this task](https://app.asana.com/0/1205683055998394/1208170657598882/f).

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208225297549515

### CODE REVIEW

**Review by alexckuhn — APPROVED** [review]
LGTM

**Comment by Greyvend** [comment]
@arthurio please check this PR, I added some more updates, beyond just the gradesReleased support.

File: pylti1p3/lineitem.py [hunk 1]
```diff
@@ -69,13 +71,14 @@ def __init__(self, lineitem: t.Optional[TLineItem] = None):
...
         self._end_date_time = lineitem.get("endDateTime")
+        self._grades_released = lineitem.get("gradesReleased")
         self._submission_review = lineitem.get("submissionReview")
         self._submission_type = lineitem.get(CANVAS_SUBMISSION_TYPE)
 
     def get_id(self) -> t.Optional[str]:
         return self._id
 
-    def set_id(self, value: str) -> "LineItem":
+    def set_id(self, value: t.Optional[str]) -> "LineItem":
```
Comment (arthurio) [inline pylti1p3/lineitem.py]: Can we do the same `str | None` notation as in the other PR?

File: pylti1p3/lineitem.py [hunk 2]
```diff
@@ -221,16 +233,25 @@ def set_submission_type(self, _type: str, external_tool_url: t.Optional[str]) ->
...
+        Prepare the line item data in the format required by the LTI-AGS specification.
+
+        The final dictionary is then converted to a JSON string and then sent via the API.
+        Any missing values are explicitly set to `None` in the dictionary and then converted to
+        `null` values in the JSON.
+        https://www.imsglobal.org/spec/lti-ags/v2p0/#updating-a-line-item
+        https://www.imsglobal.org/spec/lti-ags/v2p0/openapi/#/default
+        """
         data = {
-            "id": self._id if self._id else None,
```
Comment (arthurio) [inline pylti1p3/lineitem.py:225]: So it relies only on the `resourceId` to match and do the `PUT` operation?

**Review by arthurio — APPROVED** [review]
Thanks for checking the documentation and getting things sorted out, the changes look good to me, just a nit about typing.

File: pylti1p3/lineitem.py (hunk 1 above)
Comment (Greyvend) [reply pylti1p3/lineitem.py]: Yea, sure. Good point.

File: pylti1p3/lineitem.py (hunk 2 above)
Comment (Greyvend) [reply pylti1p3/lineitem.py:225]: The `id` is set in the URL we're issuing a `PUT` request to. The LTI standard specifies that you can put it in the payload, but there's zero reason for that.

**Pull request merged**